web/src/realtime-info/reducer.js

```diff
@@ -0,0 +1,23 @@
+import { REMOVE_PEER_STATE, UPDATE_PEER_STATE } from './actions'
+// const defaultState = {
+//   isOpen: false,
+//   goalAddress: null,
+// }
+const defaultState = {}
+
+export default function(state = defaultState, action) {
+  const { payload, type } = action
+  switch (type) {
+    case UPDATE_PEER_STATE:
+      return {
+        ...state,
+        [payload.agentPubKey]: payload
+      }
+    case REMOVE_PEER_STATE:
+      state = state
+      delete state[payload]
+      return state
```

Comment:
I left a comment previously about immutability and the situation of removing a key from an object, if you can apply that here